### PR TITLE
53-support-basic-current-location-indicator

### PR DESCRIPTION
**Pull request by JulianBissekkou**

### CODE REVIEW

File: arcgis_map_sdk/lib/src/arcgis_location_display.dart [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+  Future<void> updateLocation(UserPosition position) {
+    _assertAttached();
+    return ArcgisMapPlatform.instance
+        .updateLocationDisplaySourcePositionManually(
+      _mapId!,
+      position,
+    );
+  }
+}
```
Comment (JulianBissekkou) [inline arcgis_map_sdk/lib/src/arcgis_location_display.dart:17]: The use case for manual location displays is that an app might have an own location stream from a different source (such as geolocator) with specific settings.
Those locations are then process by my app and I would like to display the exact location that my app processes. For that I don't want arcgis to create a location client that fetches the position again, I want to show the location that I already retrieved.

File: arcgis_map_sdk_android/android/src/main/kotlin/dev/fluttercommunity/arcgis_map_sdk_android/ArcgisMapView.kt [hunk 2]
```diff
@@ -110,19 +114,172 @@ internal class ArcgisMapView(
...
+        result: MethodChannel.Result
+    ) {
+        val active = call.arguments as? Boolean
+        if (active == null) {
+            result.error("missing_data", "Invalid arguments.", null)
+            return
+        }
+
+        mapView.locationDisplay.isUseCourseSymbolOnMovement = active
+        result.success(true)
```
Comment (passsy) [inline arcgis_map_sdk_android/android/src/main/kotlin/dev/fluttercommunity/arcgis_map_sdk_android/ArcgisMapView.kt:216]: always use a try-catch, anything might fail

File: arcgis_map_sdk_android/android/src/main/kotlin/dev/fluttercommunity/arcgis_map_sdk_android/ArcgisMapView.kt [hunk 3]
```diff
@@ -110,19 +114,172 @@ internal class ArcgisMapView(
...
+                null
+            )
+            return
+        }
+
+        val optionParams = call.arguments as Map<String, Any>
+        val position = optionParams.parseToClass<UserPosition>()
+
+        dataSource.setNewLocation(position)
+        result.success(true)
```
Comment (passsy) [inline arcgis_map_sdk_android/android/src/main/kotlin/dev/fluttercommunity/arcgis_map_sdk_android/ArcgisMapView.kt:238]: catch potential errors

File: arcgis_map_sdk_android/android/src/main/kotlin/dev/fluttercommunity/arcgis_map_sdk_android/ArcgisMapView.kt [hunk 4]
```diff
@@ -110,19 +114,172 @@ internal class ArcgisMapView(
...
+                result.success(true)
+            }
+
+            "system" -> {
+                mapView.locationDisplay.locationDataSource = AndroidLocationDataSource(context)
+                result.success(true)
+            }
+
+            else -> result.error("invalid_data", "Unknown data source type ${call.arguments}", null)
+        }
```
Comment (passsy) [inline arcgis_map_sdk_android/android/src/main/kotlin/dev/fluttercommunity/arcgis_map_sdk_android/ArcgisMapView.kt:263]: catch potential errors

**Review by passsy — APPROVED** [review]

**Pull request merged**